### PR TITLE
[core] improve iTunes Transporter code and tests

### PR DESCRIPTION
**Pull request by janpio**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
The code for iTunes Transporter and its tests were confusing in several places.

### Description

- `shell_escaped_password` does strange, but necessary things on Mac. Make sure only happens on Mac only.
- `itunes_transporter_spec.rb` had a super scary hardcoded password string. Extract that to an instance variable and some additional code that is executed where necessary.
- allow `itunes_transporter_spec.rb` tests to also run on other platforms, remove "pending"

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Pull Request is Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: fastlane_core/spec/itunes_transporter_spec.rb [hunk 1]
```diff
@@ -2,14 +2,25 @@
...
 describe FastlaneCore do
+  before(:each) do
+    @password = "!> p@$s_-+=w'o%rd\"&#*<"
+  end
+
   describe FastlaneCore::ItunesTransporter do
     def shell_upload_command(provider_short_name = nil)
+      password = @password.dup
+      password = password.shellescape
+      if FastlaneCore::Helper.mac?
```
[the remark this answers is not in the document]
Comment (janpio) [reply fastlane_core/spec/itunes_transporter_spec.rb]: this just reproduces what was done in `itunes_transporter.rb`

File: fastlane_core/spec/itunes_transporter_spec.rb [hunk 2]
```diff
@@ -20,12 +31,19 @@ def shell_upload_command(provider_short_name = nil)
     end
 
     def shell_download_command(provider_short_name = nil)
+      password = @password.dup
+      password = password.shellescape
+      if FastlaneCore::Helper.mac?
```
[the remark this answers is not in the document]
Comment (janpio) [reply fastlane_core/spec/itunes_transporter_spec.rb]: this just reproduces what was done in `itunes_transporter.rb`

File: fastlane_core/spec/itunes_transporter_spec.rb [hunk 3]
```diff
@@ -2,14 +2,25 @@
 require 'credentials_manager'
 
 describe FastlaneCore do
+  before(:each) do
+    @password = "!> p@$s_-+=w'o%rd\"&#*<"
```
[the remark this answers is not in the document]
Comment (janpio) [reply fastlane_core/spec/itunes_transporter_spec.rb]: yep, I didn't know this existed ;)

File: fastlane_core/lib/fastlane_core/itunes_transporter.rb [hunk 4]
```diff
@@ -203,21 +203,23 @@ def handle_error(password)
...
-      # we use the 'do' version of gsub, because two-param version interprets the replace text as a pattern and does the wrong thing
-      password = Shellwords.escape(password).gsub("\\'") do
-        "'\"\\'\"'"
-      end
-
-      # wrap the fully-escaped password in single quotes, since the transporter expects a escaped password string
-      # (which must be single-quoted for the shell's benefit [on non-Windows platforms])
-      password = "'" + password + "'" unless Helper.windows?
+      password = password.shellescape
+      if Helper.mac?
```
[the remark this answers is not in the document]
Comment (janpio) [reply fastlane_core/lib/fastlane_core/itunes_transporter.rb]: There are no tests for why this is necessary and I couldn't figure out what it is _really_ doing and most important _why_. Removing it didn't do any damage, but as it was added for a reason and there were references to mac I left it in for Mac only. Very open for future removal of this - but I didn't want to be the one and not with this PR.

Agree?

File: fastlane_core/spec/itunes_transporter_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshdholtz) [reply fastlane_core/spec/itunes_transporter_spec.rb]: `itunes_transporter` did originally have `unless Helper.windows?` in   `password = "'" + password + "'" unless Helper.windows?` which is why I asked if that is what it should be 🙃

[the remark this answers is not in the document]
Comment (janpio) [reply fastlane_core/spec/itunes_transporter_spec.rb]: I added that to make it work on Windows some time ago:
https://github.com/fastlane/fastlane/commit/2990de41df3de3a1fa4299441f18d235f56a787f
🐰

File: fastlane_core/spec/itunes_transporter_spec.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (janpio) [reply fastlane_core/spec/itunes_transporter_spec.rb]: I replaced this now, and fixed a few other uglynesses along the way.

**Review by joshdholtz — APPROVED** [review]

**Pull request merged**